Processor.py

```diff
@@ -58,10 +53,160 @@ def decode_instruction(self, bin_instruction):
         # verificar qual a instrução e decidir quantos bytes a mais vai ter que ler (usando o read_memo)
         # cada instrução pode ter de 1 a 3 bytes (instrução no primeiro byte e valores no segundo/terceiro)
 
+        #---------------------- ADC Instruction----------------------------------
         if bin_instruction == int('69', 16): # ADC Immediate
-            immediate = self.read_memo()
-            return methods._adc(self, immediate)
+            value = self.read_memo() #immediate
+            return methods._adc(self, value)
+
+        elif bin_instruction == int('65', 16): # ADC Zero Page
+            zero_position = self.read_memo()
+            value = self.memory.read_memo(zero_position)
+            return methods._adc(self, value)
+
+        elif bin_instruction == int('75', 16): # ADC Zero Page,X
+            zero_position = self.read_memo() + self.X.value

```

Comment:
This can be more than 255, need to check for overflow.
```python
zero_position = self.read_memo() + self.X.value
if zero_position > 255:
    zero_position -= 256
```